lib/postgrex/protocol.ex

```diff
@@ -763,6 +765,86 @@ defmodule Postgrex.Protocol do
     end
   end
 
+  ## check_target_server_type
+
+  defp check_target_server_type(s, %{target_server_type: :any} = status, buffer),
+    do: check_target_server_type_done(s, status, buffer)
+
+  defp check_target_server_type(s, status, buffer),
+    do: check_target_server_type_send(s, status, buffer)
+
+  defp check_target_server_type_send(s, status, buffer) do
+    msg = msg_query(statement: "show transaction_read_only")
+
+    case msg_send(s, msg, buffer) do
+      :ok ->
+        check_target_server_type_recv(s, status, buffer)
+
+      {:disconnect, err, s} ->
+        check_target_server_type_fail(s, err, status)
+    end
+  end
+
+  defp check_target_server_type_recv(s, %{target_server_type: expected_server_type} = status, buffer) do
+    case msg_recv(s, :infinity, buffer) do
+      {:ok, msg_row_desc(fields: fields), buffer} ->
+        {[_text_type_oid = 25], ["transaction_read_only"]} = columns(fields)
```

Comment:
I'm not sure about this check on the oid for the type `text`. Is it guaranteed to be 25?